alf/algorithms/off_policy_algorithm.py

```diff
@@ -104,17 +104,21 @@ def _train_iter_off_policy(self):
         if not config.update_counter_every_mini_batch:
             alf.summary.get_global_counter().add_(1)
 
-        with record_time("time/unroll"):
-            training_info = self.unroll(config.unroll_length)
-            self.summarize_rollout(training_info)
-            self.summarize_metrics()
+        with torch.set_grad_enabled(config.unroll_with_grad):
+            with record_time("time/unroll"):
+                training_info = self.unroll(config.unroll_length)
+                self.summarize_rollout(training_info)
+                self.summarize_metrics()
 
         steps = self.train_from_replay_buffer(update_global_counter=True)
 
         with record_time("time/after_train_iter"):
-            training_info = training_info._replace(
-                rollout_info=(), info=training_info.rollout_info)
-            self.after_train_iter(training_info)
+            if config.unroll_with_grad:
+                training_info = training_info._replace(
+                    rollout_info=(), info=training_info.rollout_info)
+                self.after_train_iter(training_info)
+            else:
+                self.after_train_iter()  # only off-policy training
```

Comment:
need to update the comment for after_train_iter about when training_info is None